sktime/performance_metrics/forecasting/_classes.py

```diff
@@ -1830,6 +1846,177 @@ class GeometricMeanAbsoluteError(BaseForecastingErrorMetricFunc):
 
     func = geometric_mean_absolute_error
 
+    def __init__(
+        self,
+        multioutput="uniform_average",
+        multilevel="uniform_average",
+    ):
+        super().__init__(multioutput=multioutput, multilevel=multilevel)
+
+    def _compute_pseudo_values(self, y_true, y_pred):
+        """Compute the jackknife pseudo-values.
+
+        Note
+        ----------
+        This private method computes the jackknife pseudo-values
+        for the GMAE metric. The pseudo-values are used to estimate
+        the influence of each observation on the overall metric.
+
+        Parameters
+        ----------
+        y_true : pd.Series or pd.DataFrame
+            Ground truth (correct) target values.
+            y can be in one of the following formats:
+            Series scitype: pd.DataFrame
+            Panel scitype: pd.DataFrame with 2-level row MultiIndex
+            Hierarchical scitype: pd.DataFrame with 3 or more level row MultiIndex
+        y_pred : pd.Series or pd.DataFrame
+            Forecasted values to evaluate
+            must be of same format as y_true, same indices and columns if indexed
+
+        Returns
+        -------
+        pseudo_values : pd.Series or pd.DataFrame
+            Jackknife pseudo-values for the GMAE metric.
+            - If self.multioutput="uniform_average", returns a pd.Series where each
+              entry is the pseudo-value for a time point, averaged over variables.
+            - If self.multioutput="raw_values", returns a pd.DataFrame where each entry
+              is the pseudo-value for a time point, for each variable.
+        """
+        raw_values = (y_true - y_pred).abs()
+
+        n = raw_values.shape[0]
+        gmae_values = []
+        for i in range(n):
+            excluded_error = np.delete(raw_values, i, axis=0)  # Exclude i-th error
+            excluded_product = np.prod(excluded_error)
```

Comment:
this leads to the right value afaik, but the computational complexity is still O(n-squared)

Here's how it could be made O(n):

take the logs first of `raw_values`, and also compute the mean of the logs. Then subtract the one from the other with an appropriate normalizing factor. At the end, apply `np.exp`.